File: 06.AndroidUsingJava/DesignPatterns/newMVP2/app/src/main/java/com/example/newmvp/favProductView/FavProductAdapter.java
package com.example.newmvp.favProductView;


import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.RatingBar;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;
import com.example.newmvp.R;
import com.example.newmvp.allProdcutView.AddProductToFav;
import com.example.newmvp.model.Product;

import java.util.List;

public class FavProductAdapter extends RecyclerView.Adapter<FavProductAdapter.ViewHolder> {
    private final Context context;
    private List<Product> values;
    private static final String TAG = "RecycleView";
    AddProductToFav listener;
    public FavProductAdapter(Context m_context , AddProductToFav listener){
        context = m_context;
        this.listener = listener;
    }
    public void setList(List<Product> updatedList){
        values = updatedList;
    }

    @NonNull
    @Override
    public FavProductAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View v = inflater.inflate(R.layout.row_content_favproducts,  parent,false);
        FavProductAdapter.ViewHolder vh = new FavProductAdapter.ViewHolder(v);
        return vh;
    }

    @Override
    public void onBindViewHolder(@NonNull FavProductAdapter.ViewHolder holder, @SuppressLint("RecyclerView") int position) {

        holder.title.setText(values.get(position).getTitle());
        holder.brand.setText(values.get(position).getBrand());
        holder.description.setText(values.get(position).getDescription());
        holder.price.setText(String.valueOf(values.get(position).getPrice()));
        holder.rating.setRating((float)values.get(position).getRate());
        Glide.with(context).load(values.get(position).getImgUrl()).apply(new RequestOptions().override(200,200))
                .placeholder(R.drawable.ic_launcher_background).error(R.drawable.ic_launcher_foreground)
                .into(holder.thumbnail);
        holder.rating.setIsIndicator(true);
        holder.delete.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.onProductClickListener(values.get(position));
            }
        });
    }

    @Override
    public int getItemCount() {
        return values.size();
    }



    public class ViewHolder extends RecyclerView.ViewHolder{
        public ImageView thumbnail;
        public TextView title;
        public TextView brand;
        public TextView price;
        public TextView description;
        public RatingBar rating;
        public View layout;
        public Button delete;
        public ViewHolder(View itemView){
            super(itemView);
            layout = itemView;
            title = (TextView) layout.findViewById(R.id.titleID);
            brand = (TextView) layout.findViewById(R.id.brandID);
            price = (TextView) layout.findViewById(R.id.priceID);
            description = (TextView) layout.findViewById(R.id.descriptionID);
            rating = (RatingBar) layout.findViewById(R.id.startsID);
            thumbnail = (ImageView) layout.findViewById(R.id.imageID);
            delete = (Button) layout.findViewById(R.id.btnRemovID);
        }


    }
}

File: AndroidUsingKotlin/Session_7/MVVM/app/src/main/java/com/example/mvvm/FavProducts/FavProductAdapter.kt
package com.example.mvvm.FavProducts

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.android_kotlin.OnProductFavClickListener
import com.example.android_kotlin.ProductsDiffUtil
import com.example.mvvm.Product
import com.example.mvvm.R


class FavProductAdapter(private val listener: OnProductFavClickListener) : ListAdapter<Product, FavProductAdapter.ViewHolder>(
    ProductsDiffUtil()
) {
    class ViewHolder(private val item: View) : RecyclerView.ViewHolder(item) {
        val productName: TextView = item.findViewById(R.id.editTitle2)
        val imageView: ImageView = item.findViewById(R.id.imageView3)
        val price: TextView = item.findViewById(R.id.editPrice2)
        val brand : TextView = item.findViewById(R.id.editBrand2)
        var btnDel : Button  = item.findViewById(R.id.btnDel)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val inflater: LayoutInflater = parent.context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val view = inflater.inflate(R.layout.fav_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val currentProduct = getItem(position)
        holder.productName.text = currentProduct.title
        holder.price.text = currentProduct.price
        holder.brand.text = currentProduct.brand

        Glide.with(holder.imageView.context)
            .load(currentProduct.thumbnail)
            .into(holder.imageView)
        holder.btnDel.text = "Delete"  // Ensure the button text is "Delete"

        holder.btnDel.setOnClickListener {
            listener.onProductClick(currentProduct)
        }


    }
}
